packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -386,5 +410,12 @@ public class RealmConfiguration private constructor(
             }
             return schemaVersion
         }
+
+        private fun validateEncryptionKey(encryptionKey: ByteArray): ByteArray {
+            if (encryptionKey.size != Realm.ENCRYPTION_KEY_LENGTH) {
+                throw IllegalArgumentException("The provided key must be ${Realm.ENCRYPTION_KEY_LENGTH} bytes.")
```

Comment:
```suggestion
                throw IllegalArgumentException("The provided key must be ${Realm.ENCRYPTION_KEY_LENGTH} bytes. The provided key was ${encryptionKey.size} bytes.")
```

We should always try to give as much information as possible in these crash messages. It becomes much easier to debug.